pkg/feature/api/flag_trigger.go

```diff
@@ -17,62 +17,681 @@ package api
 
 import (
 	"context"
+	"encoding/base64"
+	"errors"
+	"fmt"
+	"net/url"
+	"strconv"
 
+	"go.uber.org/zap"
+	"google.golang.org/genproto/googleapis/rpc/errdetails"
+
+	"github.com/bucketeer-io/bucketeer/pkg/feature/domain"
+	v2fs "github.com/bucketeer-io/bucketeer/pkg/feature/storage/v2"
+	"github.com/bucketeer-io/bucketeer/pkg/locale"
+	"github.com/bucketeer-io/bucketeer/pkg/log"
+	"github.com/bucketeer-io/bucketeer/pkg/storage/v2/mysql"
+	"github.com/bucketeer-io/bucketeer/pkg/uuid"
+	accountproto "github.com/bucketeer-io/bucketeer/proto/account"
 	featureproto "github.com/bucketeer-io/bucketeer/proto/feature"
 )
 
+const maskURI = "********"
+
 func (s *FeatureService) CreateFlagTrigger(
 	ctx context.Context,
 	request *featureproto.CreateFlagTriggerRequest,
 ) (*featureproto.CreateFlagTriggerResponse, error) {
-	return &featureproto.CreateFlagTriggerResponse{}, nil
+	localizer := locale.NewLocalizer(ctx)
+	_, err := s.checkRole(ctx, accountproto.Account_EDITOR, request.EnvironmentNamespace, localizer)
+	if err != nil {
+		return nil, err
+	}
+	if err = validateCreateFlagTriggerCommand(request.CreateFlagTriggerCommand, localizer); err != nil {
+		s.logger.Info(
+			"Invalid argument",
+			log.FieldsFromImcomingContext(ctx).AddFields(
+				zap.Error(err),
+				zap.String("environmentNamespace", request.EnvironmentNamespace),
+			)...,
+		)
+		return nil, err
+	}
+	triggerID, err := uuid.NewUUID()
+	if err != nil {
+		s.logger.Error(
+			"Failed to generate trigger id",
+			log.FieldsFromImcomingContext(ctx).AddFields(
+				zap.Error(err),
+				zap.String("environmentNamespace", request.EnvironmentNamespace),
+			)...,
+		)
+		return nil, err
+	}
+	id := triggerID.String()
+	triggerUUID, err := uuid.NewUUID()
+	if err != nil {
+		s.logger.Error(
+			"Failed to generate trigger uuid",
+			log.FieldsFromImcomingContext(ctx).AddFields(zap.Error(err))...,
+		)
+		return nil, err
+	}
+	flagTriggerUuid := triggerUUID.String()
+	flagTrigger := domain.NewFlagTrigger(
```

Comment:
Please implement the command layer to keep consistency with the other APIs.
Also, you need to implement the logic to publish the domain events which is done in the domain package.

Implement the command layer in all the APIs.
Beware that all actions must publish a domain event so that we can track the changes in the audit log.

E.g.

### Command
https://github.com/bucketeer-io/bucketeer/blob/main/pkg/experiment/command/experiment.go#L50-L73

### Domain event
https://github.com/bucketeer-io/bucketeer/blob/main/pkg/experiment/command/experiment.go#L110-L113

Reply:
Oh, I forgot about the logic for the audit log.
I will implement it similar to how it is done in the other service.